packages/devtools_app/lib/src/timeline/timeline_model.dart

```diff
@@ -1042,6 +1043,11 @@ class AsyncTimelineEvent extends TimelineEvent {
       asyncId == eventWrapper.event.id,
       'asyncId = $asyncId, but endEventId = ${eventWrapper.event.id}',
     );
+    if (endTraceEventJson != null) {
+      // This event has already ended and [eventWrapper] is a duplicate trace
+      // event.
+      return;
+    }
     if (name == eventWrapper.event.name && endTraceEventJson == null) {
```

Comment:
the endTraceEventJson == null check here now seems obsolete.
you could add it back as an assert if you wanted.

Reply:
removed